### PR TITLE
ScalaBuffer improvements

### PR DESCRIPTION
**Pull request by tindzk**

### CODE REVIEW

File: html/src/main/scala/org/hyperscala/html/tag/Head.scala [hunk 1]
```diff
@@ -135,6 +135,11 @@ class Head extends Container[HeadChild] with HTMLChild with HTMLTag {
     case None => contents += new Link(href = url)
   }
 
+  def meta(name: String): scala.Option[String] =
+    contents.collectFirst {
+      case m: Meta if m.name().toLowerCase == name => m.content()
+    }
+
   def meta(name: String, content: String) = {
     contents.collectFirst {
```
Comment (darkfrog26) [inline html/src/main/scala/org/hyperscala/html/tag/Head.scala]: Probably would be beneficial to change this to use the new `meta(String)` method.

File: html/src/main/scala/org/hyperscala/html/tag/Video.scala [hunk 2]
```diff
@@ -10,7 +10,7 @@ import org.hyperscala.html.constraints._
  * NOTE: This file has been generated. Do not modify directly!
  * @author Matt Hicks <matt@outr.com>
  */
-class Video extends Container[Source] with BodyChild with HTMLTag {
+class Video extends Container[BodyChild] with BodyChild with HTMLTag {
```
Comment (darkfrog26) [inline html/src/main/scala/org/hyperscala/html/tag/Video.scala]: `Source` was a more restrictive type than `BodyChild`. There should be no circumstance where anything is added to `Video` that is not `Source`.

File: web/src/main/scala/org/hyperscala/io/HTMLToScala.scala [hunk 3]
```diff
@@ -32,17 +33,19 @@ object HTMLToScala {
     parent.read(body)                                  // Read the new data back in
   }
 
-  def toScala(page: Webpage, packageName: String, className: String) = {
+  def toScala(page: tag.HTML, packageName: Option[String], className: String): String = {
```
Comment (darkfrog26) [inline web/src/main/scala/org/hyperscala/io/HTMLToScala.scala]: If you're going to change the type here you should also change the variable name `page` to something like `html`.

**Comment by darkfrog26** [comment]
As you're making changes to this it would probably be worthwhile to start writing tests to verify the functionality works as expected.

File: html/src/main/scala/org/hyperscala/html/tag/Video.scala (hunk 2 above)
Comment (tindzk) [inline html/src/main/scala/org/hyperscala/html/tag/Video.scala]: I've encountered an HTML file that contains a paragraph within a ``<video>`` tag:

```html
<video ...>
  <p>To view this video please enable JavaScript.</p>
</video>
```

File: html/src/main/scala/org/hyperscala/html/tag/Head.scala (hunk 1 above)
Comment (tindzk) [inline html/src/main/scala/org/hyperscala/html/tag/Head.scala]: This will not work as ``meta(name)`` would need to return ``Option[Meta]``.

Comment (darkfrog26) [inline html/src/main/scala/org/hyperscala/html/tag/Head.scala]: ah, you're right.

File: html/src/main/scala/org/hyperscala/html/tag/Video.scala (hunk 2 above)
Comment (darkfrog26) [inline html/src/main/scala/org/hyperscala/html/tag/Video.scala]: Ugh, sometimes I hate HTML. :o

**Pull request merged**